File: app/discover/scan_region.rb
#!/usr/bin/env ruby

# scan a region for projects, availability zones and aggregates

require 'singleton'
require_relative 'db_fetch_projects'
require_relative 'db_fetch_aggregates'
require_relative 'db_fetch_availability_zones'
require_relative 'db_fetch_availability_zones'
require_relative 'scan_aggregate'
require_relative 'scan_availability_zone'

class ScanRegion < Scanner
  include Singleton
  
  def initialize()
    super([
      {
        :type => "aggregate",
        :fetcher => DbFetchAggregates.new(),
        :children_scanner => ScanAggregate.instance
      },
      {
        :type => "availability zone",
        :fetcher => DbFetchAvailabilityZones.new(),
        :children_scanner => ScanAvailabilityZone.instance
      }
    ])
  end
  
end

File: app/discover/scan_region.py
# scan a region for availability zones and aggregates
from discover.folder_fetcher import FolderFetcher
from discover.scanner import Scanner
from utils.singleton import Singleton


class ScanRegion(Scanner, metaclass=Singleton):
    def __init__(self):
        super(ScanRegion, self).__init__([
            {
                "type": "aggregates_folder",
                "fetcher": FolderFetcher("aggregates", "region"),
                "children_scanner": "ScanAggregatesRoot"
            },
            {
                "type": "network",
                "fetcher": "ApiFetchNetworks",
                "children_scanner": "ScanNetwork"
            },
            {
                "type": "port",
                "fetcher": "ApiFetchPorts"
            }
        ])
